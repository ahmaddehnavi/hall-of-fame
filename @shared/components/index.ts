export * from './layout/Col'
export * from './layout/Row'
export * from './screen/Screen'
export * from './backhandler/MultiBackHandler'
export * from './touchable/Touchable'
export * from './button/Button'
export * from './list/FlatList'
export * from './empty/EmptyView'
export * from './loading/LoadingView'
export * from './error/ErrorView'
export * from './async-view/AsyncView'