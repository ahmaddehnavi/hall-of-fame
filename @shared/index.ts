export * from './services'
export * from './utils'
export * from './components'
export * from './resource/Resource'
