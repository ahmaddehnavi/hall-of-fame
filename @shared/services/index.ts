export * from './base'
export * from './navigation'
export * from './theme'

