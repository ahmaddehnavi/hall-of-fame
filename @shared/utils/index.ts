export * from './di/DIUtil'
export * from './async/AsyncUtil'