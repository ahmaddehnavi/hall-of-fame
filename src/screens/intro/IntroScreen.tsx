import {Col, INavigationService, PXInject, PXInjectedNavigationServiceProps, Screen, Touchable} from '@shared';
import autobind from 'autobind-decorator';
import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {WelcomeScreen} from '../welcome/WelcomeScreen';

export function IntroScreenComponent(props: { onPress: () => void }) {
    return (
        <Screen>
            <Touchable onPress={props.onPress}>
                <Col style={styles.container}>
                    <Text>IntroScreen</Text>
                </Col>
            </Touchable>
        </Screen>
    )
}

export type IntroScreenProps =
    PXInjectedNavigationServiceProps

@PXInject('$navigation')
export class IntroScreen extends React.Component<IntroScreenProps> {
    static readonly ROUTE_NAME = 'IntroScreen';

    static start(nav: INavigationService) {
        nav.navigate(this.ROUTE_NAME);
    }

    static resetTo(nav: INavigationService) {
        nav.reset(this.ROUTE_NAME);
    }

    @autobind
    handleOnPress() {
        WelcomeScreen.start(this.props.$navigation);
    }

    render() {
        return (
            <IntroScreenComponent onPress={this.handleOnPress}/>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center'
    }
});