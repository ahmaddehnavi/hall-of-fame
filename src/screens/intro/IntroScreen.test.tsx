import {PXNavigationService} from '@shared';
import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {IntroScreen} from './IntroScreen';

it('renders correctly', () => {
    let tree = renderer.create(
        <IntroScreen
            $navigation={new PXNavigationService()}
        />
    );
    expect(tree).toMatchSnapshot();
});

