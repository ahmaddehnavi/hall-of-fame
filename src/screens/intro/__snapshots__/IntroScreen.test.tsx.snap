// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 184, 212, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      bounces={false}
      data={
        Array [
          Object {
            "backgroundColor": "#00b8d4",
            "image": "",
            "key": "slide1",
          },
          Object {
            "backgroundColor": "#64dd17",
            "image": "",
            "key": "slide2",
          },
          Object {
            "backgroundColor": "#ffd600",
            "image": "",
            "key": "slide13",
          },
          Object {
            "backgroundColor": "#ff6d00",
            "image": "",
            "key": "slide14",
          },
          Object {
            "backgroundColor": "#ff0000",
            "image": "",
            "key": "slide15",
          },
        ]
      }
      disableVirtualization={false}
      extraData={750}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onDone={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      paginationStyle={null}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showDoneButton={true}
      showNextButton={true}
      showsHorizontalScrollIndicator={false}
      slides={
        Array [
          Object {
            "backgroundColor": "#00b8d4",
            "image": "",
            "key": "slide1",
          },
          Object {
            "backgroundColor": "#64dd17",
            "image": "",
            "key": "slide2",
          },
          Object {
            "backgroundColor": "#ffd600",
            "image": "",
            "key": "slide13",
          },
          Object {
            "backgroundColor": "#ff6d00",
            "image": "",
            "key": "slide14",
          },
          Object {
            "backgroundColor": "#ff0000",
            "image": "",
            "key": "slide15",
          },
        ]
      }
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 1334,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source=""
              style={
                Object {
                  "flex": 1,
                  "width": "90%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 1334,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source=""
              style={
                Object {
                  "flex": 1,
                  "width": "90%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 1334,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source=""
              style={
                Object {
                  "flex": 1,
                  "width": "90%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 1334,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source=""
              style={
                Object {
                  "flex": 1,
                  "width": "90%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 1334,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source=""
              style={
                Object {
                  "flex": 1,
                  "width": "90%",
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "bottom": 16,
            "left": 16,
            "position": "absolute",
            "right": 16,
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 16,
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, .9)",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 12,
                },
                null,
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
